src/net.cpp

```diff
@@ -1019,6 +1021,9 @@ void CConnman::CreateNodeFromAcceptedSocket(std::unique_ptr<Sock>&& sock,
     }
 
     const bool inbound_onion = std::find(m_onion_binds.begin(), m_onion_binds.end(), addr_bind) != m_onion_binds.end();
+    CNodeOptions node_opts;
+    node_opts.prefer_evict = discouraged;
```

Comment:
`prefer_evict` is set here, but later ignored by the `CNode` constructor. Maybe this was forgotten:

```diff
--- i/src/net.cpp
+++ w/src/net.cpp
@@ -2733,12 +2733,13 @@ CNode::CNode(NodeId idIn,
       m_sock{sock},
       m_connected{GetTime<std::chrono::seconds>()},
       addr{addrIn},
       addrBind{addrBindIn},
       m_addr_name{addrNameIn.empty() ? addr.ToStringIPPort() : addrNameIn},
       m_inbound_onion{inbound_onion},
+      m_prefer_evict{node_opts.prefer_evict},
       nKeyedNetGroup{nKeyedNetGroupIn},
       id{idIn},
       nLocalHostNonce{nLocalHostNonceIn},
       m_conn_type{conn_type_in},
       m_i2p_sam_session{std::move(node_opts.i2p_sam_session)}
 {
```